packages/lambda/src/lib/store/index.js

```diff
@@ -34,37 +35,22 @@ export class ClaimStorage {
   }
 
   /** @param {import('@web3-storage/content-claims/server/api').Claim} claim */
-  async put ({ claim, bytes, content, expiration }) {
-    const hasExpiration = expiration && isFinite(expiration)
-    const cidstr = claim.toString(base32)
+  async put (claim) {
     await Promise.all([
-      retry(() => {
-        const cmd = new PutObjectCommand({
-          Bucket: this.#bucket.bucketName,
-          Key: `${cidstr}/${cidstr}.car`,
-          Body: bytes
-        })
-        return this.#bucket.s3Client.send(cmd)
-      }, {
-        minTimeout: 100,
-        onFailedAttempt: err => console.warn(`failed S3 put for: ${cidstr}`, err)
-      }),
-      retry(() => {
-        const cmd = new UpdateItemCommand({
-          TableName: this.#table.tableName,
-          Key: marshall({
-            claim: claim.toString(),
-            content: base58btc.encode(content.bytes)
-          }),
-          ExpressionAttributeValues: marshall({ ':ex': hasExpiration ? expiration : 0 }),
-          UpdateExpression: 'SET expiration=:ex'
-        })
-        return this.#table.dynamoClient.send(cmd)
-      }, {
-        minTimeout: 100,
-        onFailedAttempt: err => console.warn(`failed DynamoDB update for: ${cidstr}`, err)
-      })
+      storeClaimBytes(claim, this.#bucket),
+      upsertClaim(claim, this.#table)
     ])
+    const { content, value } = claim
+    if (value.can === 'assert/equals') {
+      const equivalent = value.nb.equals.multihash
+      if (Bytes.equals(content.bytes, equivalent.bytes)) {
+        // the multihash matches so this claim will appear in queries for either CID already.
+        return
+      }
+      // also add the claim with the `equals` cid as the `content` cid,
+      // so we can provide the equivalent claim for look ups for either.
+      await upsertClaim({ ...claim, content: equivalent }, this.#table)
```

Comment:
I wonder if we should add any metadata to this? Otherwise, might get complex if we in the future want to change something and not know what was added on the other way. Thoughts?

Reply:
I'm in favour of adding the capability name to the dynamo record, so we could in future we could let user ask for "just `assert/equals` or whichever flavour they want, and it has the side benefit of letting us reduce the number of records we'd have to explore to back out this choice, without adding a explicit "this is one of those reverse claims"

Reply:
https://github.com/web3-storage/content-claims/issues/26